Sources/Overlays/_Testing_CoreGraphics/Attachments/Attachment+AttachableAsCGImage.swift

```diff
@@ -0,0 +1,147 @@
+  ///     If this type does not conform to [`UTType.image`](https://developer.apple.com/documentation/uniformtypeidentifiers/uttype-swift.struct/image),
+  ///     the result is undefined. Pass `nil` to let the testing library decide
+  ///     which image format to use.
+  ///   - encodingQuality: The encoding quality to use when encoding the image.
+  ///     If the image format used for encoding (specified by the `contentType`
+  ///     argument) does not support variable-quality encoding, the value of
+  ///     this argument is ignored.
+  ///   - sourceLocation: The source location of the call to this initializer.
+  ///     This value is used when recording issues associated with the
+  ///     attachment.
+  ///
+  /// The following system-provided image types conform to the
+  /// ``AttachableAsCGImage`` protocol and can be attached to a test:
+  ///
+  /// - [`CGImage`](https://developer.apple.com/documentation/coregraphics/cgimage)
+  @_spi(Experimental)
+  @available(_uttypesAPI, *)
+  public init<T>(
+    _ attachableValue: T,
+    named preferredName: String? = nil,
+    as contentType: UTType?,
+    encodingQuality: Float = 1.0,
+    sourceLocation: SourceLocation = #_sourceLocation
+  ) where AttachableValue == _AttachableImageContainer<T> {
+    self.init(attachableValue: attachableValue, named: preferredName, as: contentType, encodingQuality: encodingQuality, sourceLocation: sourceLocation)
+  }
+
+  /// Initialize an instance of this type that encloses the given image.
+  ///
+  /// - Parameters:
+  ///   - attachableValue: The value that will be attached to the output of
+  ///     the test run.
+  ///   - preferredName: The preferred name of the attachment when writing it
+  ///     to a test report or to disk. If `nil`, the testing library attempts
+  ///     to derive a reasonable filename for the attached value.
+  ///   - encodingQuality: The encoding quality to use when encoding the image.
+  ///     If the image format used for encoding (specified by the `contentType`
+  ///     argument) does not support variable-quality encoding, the value of
+  ///     this argument is ignored.
+  ///   - sourceLocation: The source location of the call to this initializer.
+  ///     This value is used when recording issues associated with the
+  ///     attachment.
+  ///
+  /// The following system-provided image types conform to the
+  /// ``AttachableAsCGImage`` protocol and can be attached to a test:
+  ///
+  /// - [`CGImage`](https://developer.apple.com/documentation/coregraphics/cgimage)
+  @_spi(Experimental)
+  public init<T>(
+    _ attachableValue: T,
+    named preferredName: String? = nil,
+    encodingQuality: Float = 1.0,
+    sourceLocation: SourceLocation = #_sourceLocation
+  ) where AttachableValue == _AttachableImageContainer<T> {
+    self.init(attachableValue: attachableValue, named: preferredName, as: nil, encodingQuality: encodingQuality, sourceLocation: sourceLocation)
```

Comment:
The `as: nil` argument here is a little unclear, which I guess is because the `contentType` property is Optional. I wonder if it'd be more clearly labeled `contentType:`?

Reply:
Sure, can rename that one. It's an internal-only initializer.